packages/wrangler/src/versions/deploy.ts

```diff
@@ -205,6 +209,30 @@ function getConfig(
 	return config;
 }
 
+/**
+ * Prompts the user for confirmation when overwriting the latest deployment, given that it's split.
+ */
+export async function confirmLatestDeploymentOverwrite(
+	accountId: string,
+	scriptName: string
+) {
+	try {
+		const latest = await fetchLatestDeployment(accountId, scriptName);
+		if (latest && latest.versions.length >= 2) {
+			logger.warn(
+				`Your latest deployment has multiple versions.\n"wrangler deploy" will upload a new version and deploy it globally immediately.`
```

Comment:
Should this also include a mention of "wrangler versions deploy" as the alternative?